common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1368,6 +1379,18 @@ def can_toggle_course_pacing(self):
         """
         return datetime.now(UTC()) <= self.start
 
+    @property
+    def gating_prerequisites(self):
+        """
+        Course content that can be used to gate other course content within this course.
+
+        Returns:
+            list: Returns a list of dicts containing the gating milestone data
+        """
+        if not self._gating_prerequisites:
```

Comment:
This keeps us from having to make a database call to retrieve gating milestone data for every content node in the course tree when rendering the studio course outline. We only need one database call to find all gating prerequisites for the whole course for a given request. I suppose this could be moved to init where we could check the `enable_subsection_gating` flag on the course, but I didn't want to make the extra database call unless we needed it.